### PR TITLE
Remove the regex filter from the mock log appender

### PR DESCRIPTION
**Pull request by thecoop**
This is for issue #90596 

The `.*` regex on the mock log appender filters out some characters that can appear in random unicode strings (depending on the random seed), breaking tests that look at log statements for some specific seeds. This filter doesn't seem to do anything in particular, so I've removed it.

### CODE REVIEW

**Comment by thecoop** [comment]
This was originally found on 7.17, but the filter has been there since 2016, so I'm not sure where this change should be applied to

File: test/framework/src/main/java/org/elasticsearch/test/MockLogAppender.java [hunk 1]
```diff
@@ -27,10 +27,10 @@ public class MockLogAppender extends AbstractAppender {
 
     private static final String COMMON_PREFIX = System.getProperty("es.logger.prefix", "org.elasticsearch.");
 
-    private List<LoggingExpectation> expectations;
+    private final List<LoggingExpectation> expectations;
 
-    public MockLogAppender() throws IllegalAccessException {
-        super("mock", RegexFilter.createFilter(".*(\n.*)*", new String[0], false, null, null), null, false);
```
Comment (pgomulka) [inline test/framework/src/main/java/org/elasticsearch/test/MockLogAppender.java:33]: are we able to parse multiline messages now? I wonder if this was the purpose of that filter.
This is used in testing only anyway, so if build is green I guess we can try this fix

**Review by pgomulka — APPROVED** [review]
LGTM, left one question

File: test/framework/src/main/java/org/elasticsearch/test/MockLogAppender.java (hunk 1 above)
Comment (thecoop) [reply test/framework/src/main/java/org/elasticsearch/test/MockLogAppender.java:33]: That was my thinking - if the builds all pass with this removed, then it's ok to remove it...

**Pull request merged**